packages/components/Menu/src/MenuItem/useMenuItem.ts

```diff
@@ -62,6 +63,9 @@ export const useMenuItem = (props: MenuItemProps): MenuItemInfo => {
 
   useHoverFocusEffect(pressable.state.hovered, componentRef);
 
+  // Track the ref and disabled props on this menu item so the MenuList can handle Home and End keypresses.
+  useMenuItemTracking(componentRef, disabled);
```

Comment:
Is this guaranteed to run in order of the children array?

Reply:
Are there scenarios where React children aren't rendered in order? I was under the impression that the children are rendered inorder in the tree, but I could be wrong.

Reply:
I'm not sure, I think it'd be worth doing some research to see if you could confirm.